GUI/GUIListPanel.cpp

```diff
@@ -824,6 +838,88 @@ void GUIListPanel::ScrollToBottom()
 }
 
 
+/// <summary>
+/// Sets whether the scroll panel scrolls in a loop or not.
+/// </summary>
+/// <param name="scrollLoop">true to scroll in a loop, false to scroll with edge stopping.</param>
+void GUIListPanel::SetMouseScrollingLoop(bool ScrollLoop)
+{
+	m_LoopMouseScroll = ScrollLoop;
+}
+
+
+/// <summary>
+/// Sets whether the list panel can be scrolled with the mouse scroll wheel.
+/// </summary>
+/// <param name="MouseScroll">true to enable scrolling, false to disable</param>
+void GUIListPanel::SetMouseScrolling(bool MouseScroll)
+{
+	m_MouseScroll = MouseScroll;
+}
+
+
+/// <summary>
+/// Perform list scrolling through the scrollbar
+/// </summary>
+/// <param name="MouseWheelChange">amount and direction of scrolling. positive to scroll up, negative to scroll down.</param>
+void GUIListPanel::ScrollBarScrolling(int MouseWheelChange)
+{
+	int newValue = 0;
+	Item* lastItem = GetItem(GetItemList()->size() - 1);
+	int avgItemHeight = static_cast<int>((GetStackHeight(lastItem) + GetItemHeight(lastItem)) / GetItemList()->size());
+	if (MouseWheelChange < 0) {
+		newValue = m_VertScroll->GetValue() - (MouseWheelChange * avgItemHeight);
+		int maxValue = GetStackHeight(lastItem) + GetItemHeight(lastItem) - m_VertScroll->GetPageSize();
+		if (maxValue < 0) {
```

Comment:
You can probably simplify this by using Limit(number, upperLimit, lowerLimit). Though it is in RTETools, which may not be included in GUI stuff, so maybe not.
There's also std::clamp but I think that's not in our cpp version https://en.cppreference.com/w/cpp/algorithm/clamp

If you've not got a built-in one available to you, the following idiom describes it and can be used pretty easily:
```
value = max(lowerLimit, min(value, upperLimit));
```

Reply:
good to know. the limit method is indeed available from that class.